cancermuts/datasources.py

```diff
@@ -1161,6 +1165,159 @@ def _get_revel_from_gc(self, mutation, gc):
+            return
+
+        if not gcs:
+            gcs = [None] * len(gms)
+        if not gms:
+            gms = [None] * len(gcs)
+
+        for gc, gm in zip(gcs, gms):
+            genome_version, chrom, pos = None, None, None
+
+            if gm and hasattr(gm, 'genome_build') and hasattr(gm, 'chr') and hasattr(gm, 'get_coord'):
+                # gm is a GenomicMutation object
+                genome_version = gm.genome_build
+                chrom = gm.chr
+                pos = str(gm.get_coord())
+            elif gc and hasattr(gc, 'genome_build') and hasattr(gc, 'chr') and hasattr(gc, 'coord_start'):
+                # gc is a GenomicCoordinate object
+                genome_version = gc.genome_build
+                chrom = gc.chr
+                pos = str(gc.coord_start)
+
+            if not all([genome_version, chrom, pos]):
+                self.log.warning(f"[REVEL] Skipping: missing genomic data for {mutation}")
+                continue
+
+            coord_index = 2 if genome_version == 'hg38' else 1 if genome_version == 'hg19' else None
+            if coord_index is None:
+                self.log.warning(f"[REVEL] Unknown genome version '{genome_version}' for {mutation}")
+                continue
+
+            key = f"{aaref}{pos}{aaalt}"
+            self.log.debug(f"[REVEL] Searching REVEL DB for key={key} in genome={genome_version}, transcript={transcript_id}")
+
+            file_path = self._database_files['revel']
+            try:
+                with open(file_path, 'r') as f:
+                    for line_num, line in enumerate(f,start=2):
+                        next(f)
+                        if transcript_id not in line:
+                            continue
+                        parts = line.strip().split(",")
+                        if len(parts) != 9:
+                            continue
+                        file_pos = parts[coord_index]
+                        file_aaref = parts[5]
+                        file_aaalt = parts[6]
+                        revel_score = parts[7]
+                        transcripts = parts[8].split(";")
+                        if (
+                            transcript_id in transcripts and
+                            file_pos == pos and
+                            file_aaref == aaref and
+                            file_aaalt == aaalt
+                        ):
+                            try:
+                                score = float(revel_score)
+                                mutation.metadata['revel_score'].append(DbnsfpRevel(source=self, score=score))
+                                self.log.info(f"[REVEL] Match found in line {line_num}: REVEL score = {score}")
+                            except Exception as e:
+                                self.log.warning(f"[REVEL] Could not parse REVEL score on line {line_num}: {e}")
+                            break
```

Comment:
I think we need to restructure this a bit 

  - as per my previous comment, I think you want to parse the REVEL file once and end up with a `pandas.DataFrame` to filter it / find your mutations
  - I tried parsing just with `pd.read_csv`, it takes ~75-80 seconds in my test. It does take a bit of memory but I think we can live with it. 
  - you shouldn't be matching on amino acid change, but on genomic mutation, since that is what we are going to score, especially for these cases, so feel free to ignore cases that don't have a genomic mutation
  - notice that you might have more than one genomic mutation per amino acid mutation (because of genetic code degeneracy) meaning that you might potentially have multiple revel scores per mutation